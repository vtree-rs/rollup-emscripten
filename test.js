'use strict';

/* eslint-env jest */

const rollupEmscripten = require('./').default;

expect.addSnapshotSerializer({
	test: value => typeof value === 'string',
	print: value => value,
});

function rtest(name, code) {
	test(name, () => rollupEmscripten({
		entry: 'test.js',
		localPrefix: 'test',
		plugins: [{
			load: () => code,
		}],
	}).then(result => result.code, error => error.message).then(result => {
		expect(result).toMatchSnapshot();
	}));
}

rtest('unsupported top-level statement', `
	sideEffect();
`);

rtest('local function referenced by exported one', `
	function x() {}

	export function y() {
		return x();
	}
`);

rtest('exported function referenced by another one', `
	export function x() {}

	export function y() {
		return x();
	}
`);

rtest('pure local variable used by exported function', `
	var x = 10 + 20 - ~30;

	export function getX() {
		return x;
	}
`);

rtest('impure local variable used by exported function', `
	var x = new ArrayBuffer(10);

	export function getX() {
		return x;
	}
`);

rtest('exported local variable using a local function', `
	export var x = getX();

	function getX() {
		return 42;
	}
`);

rtest('exporting a variable and a function with custom names', `
	var localVar = 10;

	function localFunc() {}

	export {
		localVar as exportedVar,
		localFunc as exportedFunc
	};
`);
