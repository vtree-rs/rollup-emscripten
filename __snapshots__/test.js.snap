// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exported function referenced by another one 1`] = `
Object.assign(LibraryManager.library, {
    x: function () {
    },
    y: function () {
        return _x();
    },
    y__deps: ['x']
});
`;

exports[`exported function referencing a constant 1`] = `
Object.assign(LibraryManager.library, {
    $test_answer: 42,
    getAnswer: function () {
        return test_answer;
    },
    getAnswer__deps: ['$test_answer']
});
`;

exports[`exported local variable using a local function 1`] = `
Object.assign(LibraryManager.library, {
    x: void 0,
    x__postset: '_x = test_getX()',
    $test_getX: function () {
        return 42;
    },
    x__deps: ['$test_getX']
});
`;

exports[`exporting a variable and a function with custom names 1`] = `
Object.assign(LibraryManager.library, {
    exportedVar: 10,
    exportedFunc: function () {
    }
});
`;

exports[`impure local variable used by exported function 1`] = `
Object.assign(LibraryManager.library, {
    $test_x: void 0,
    $test_x__postset: 'test_x = new ArrayBuffer(10)',
    getX: function () {
        return test_x;
    },
    getX__deps: ['$test_x']
});
`;

exports[`local function referenced by exported one 1`] = `
Object.assign(LibraryManager.library, {
    $test_x: function () {
    },
    y: function () {
        return test_x();
    },
    y__deps: ['$test_x']
});
`;

exports[`pure local variable used by exported function 1`] = `
Object.assign(LibraryManager.library, {
    $test_x: 10 + 20 - ~30,
    getX: function () {
        return test_x;
    },
    getX__deps: ['$test_x']
});
`;

exports[`unsupported top-level statement 1`] = `
Unsupported top-level statement:
{
  "type": "ExpressionStatement",
  "start": 0,
  "end": 13,
  "expression": {
    "type": "CallExpression",
    "start": 0,
    "end": 12,
    "callee": {
      "type": "Identifier",
      "start": 0,
      "end": 10,
      "name": "sideEffect"
    },
    "arguments": []
  }
}
`;
